### ISSUE TITLE
Crash if source path contains unicode character

### ISSUE THREAD

**Issue by vorg** [jmathai/elodie#167]
I'm trying to import photos exported from iPhoto and one of the folders has Islandic characters in it that causes import to crash `"2016-03 Iceland/Suðurland - Iceland, 2"`

```
./elodie.py import --destination="/Volumes/BACKUP2TB01/Photos" --source="/Volumes/BACKUP2TB01/Photos ToDo/2016-03 Iceland/Suðurland - Iceland, 2"
Traceback (most recent call last):
  File "./elodie.py", line 301, in <module>
    main()
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "./elodie.py", line 111, in _import
    trash, allow_duplicates)
  File "./elodie.py", line 38, in import_file
    _file = _decode(_file)
  File "/Users/vorg/Dev/elodie/elodie/compatability.py", line 3, in _decode
    return string.decode(encoding)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/encodings/utf_8.py", line 16, in decode
    return codecs.utf_8_decode(input, errors, True)
UnicodeEncodeError: 'ascii' codec can't encode character u'\xf0' in position 51: ordinal not in range(128)
```

**Comment by jmathai** [comment]
Are you on the latest code from the master branch? This should have been fixed last week in #162/#166. // @misterpipine

*originally posted in the wrong issue*

**Comment by jmathai** [comment]
@vorg nevermind - I can see from the stack trace that you're on the latest commit. I thought we fixed this but there still seems to be an issue. I'll have a look.

**Comment by jmathai** [comment]
@vorg do you mind seeing if the changes in gh-170 help address your issue?

**Comment by jmathai** [comment]
@misterpipine can you clarify if it does or does not work?

**Comment by misterpipine** [comment]
sorry, it doesn't

**Comment by jmathai** [comment]
But master does, correct?

**Comment by jmathai** [comment]
I also have #170 with some changes if you can verify. Also if you can send me a file which fails I can try to reproduce. jaisen(at)github-username.com

**Comment by misterpipine** [comment]
master, doesn't. The problem comes from UTF-8 filesystem, not files themselves

**Comment by jmathai** [comment]
@misterpipine I'm confused as to why the master branch fails for path names with unicode. What's on master is the latest of what we verified and wrote tests for. And I get the same response as you did so I'm not sure why I'm unable to reproduce it.

```
>>> import sys
>>> print sys.getdefaultencoding() #python
ascii
>>> print sys.getfilesystemencoding() #OS
utf-8
```

... unless I'm misunderstanding.

**Comment by phifogg** [comment]
Send you (Jaisen) an image causing this error on my mac. As mentioned in #179 i believe this is the same issue.

**Comment by phifogg** [comment]
email adress did not work :/ - and I did replace the (at) :)

**Comment by misterpipine** [comment]
@jmathai: not sure what we missed during our tests but my first correction was done in elodie.py, function _import(...)
```
...
destination = destination.decode(sys.getfilesystemencoding())
...
source = source.decode(sys.getfilesystemencoding())
...
```
This correction worked for me but we modified this initial code to make the test running fine and for Python 2 and 3 compatibility. My proposal is re-apply the correction above and see the tests later.

**Comment by jmathai** [comment]
@phifogg jaisen(at)jmathai.com

@misterpipine The `_decode` only wraps the `string.decode()` function in a python3 compatible way. I believe calling `decode()` directly without raise an error in python 3 and is why I moved it to the `compatability.py` file. I'm without a laptop for a few days so I can't test but if you see something incorrect in the changes I made that don't properly apply your patch then we should address that.

**Comment by misterpipine** [comment]
Ok @jmathai I'll have a look to our patch when I'll have some time. I also have to improve my Git skills before ;-)

**Comment by jmathai** [comment]
@misterpipine Mind seeing if this fixes the issue on your machine? https://github.com/jmathai/elodie/pull/180

**Comment by jmathai** [comment]
@phifogg confirmed that #180 fixed the issue he reported (#179).

@misterpipine I'm going to merge that and close this. We can reopen if the issue still exists for you but they seem to be duplicates.

**Comment by jmathai** [comment]
Closed by #180.

**Comment by Poludo** [comment]
z  e ez
